ext/simplexml/tests/gh12223.phpt

```diff
@@ -0,0 +1,67 @@
+--TEST--
+GH-12223: Entity reference produces infinite loop in var_dump/print_r
+--EXTENSIONS--
+simplexml
+--FILE--
+<?php
+
+$xml = <<<XML
+<?xml version="1.0"?>
+<!DOCTYPE somedoc [
+  <!ENTITY a "something">
+  <!ENTITY b "&a;">
+  <!ENTITY c "&b;">
+]>
+<somedoc>&c;</somedoc>
+XML;
+
+$sxe = simplexml_load_string($xml);
+
+var_dump($sxe);
+print_r($sxe);
+
+?>
+--EXPECT--
+object(SimpleXMLElement)#1 (1) {
+  ["c"]=>
+  object(SimpleXMLElement)#2 (1) {
+    ["c"]=>
```

Comment:
Clearly those are different object but they "seem" identical?

Reply:
This confused me too, but it is pre-existing behaviour and thus not something we can just change.
From an XML perspective it kind of makes sense: the declaration (second c) is a child of the reference (first c), etc. So that's why they show up like this.

Reply:
ACK